sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleOverviewView.groovy

```diff
@@ -42,7 +44,18 @@ class SampleOverviewView extends VerticalLayout implements HasHotbar, HasTitle,
 
         this.samplesGrid = createSamplesGrid(viewModel.samples)
         samplesGrid.setSizeFull()
-        this.addComponents(samplesGrid)
+        this.statusComboBox = new ComboBox<>("Status", Status.values().toList())
+        statusComboBox.setItemCaptionGenerator({it.toString()})
+        statusComboBox.setEmptySelectionCaption("All statuses")
+        statusComboBox.addValueChangeListener({
+            if (it.getValue()) {
+                sampleFilter.withStatus(it.getValue().toString())
+            } else {
+                sampleFilter.clearStatus()
+            }
+            samplesGrid.dataProvider.refreshAll()
+        })
+        this.addComponents(statusComboBox, samplesGrid)
```

Comment:
Maybe we can move this code into a method createStatusComboBox() or so? This would keep the init() clean